File: leetcode/0744.find-smallest-letter-greater-than-target/solution.cpp
// https://leetcode.com/problems/find-smallest-letter-greater-than-target

class Solution {
public:
    static char nextGreatestLetter(vector<char>& letters, char target) {
        auto it = std::upper_bound(letters.begin(), letters.end(), target);
        return it == letters.end() ? letters[0] : *it;
    }
};

File: leetcode/0744.find-smallest-letter-greater-than-target/solution.rs
// https://leetcode.com/problems/find-smallest-letter-greater-than-target/
impl Solution {
    pub fn next_greatest_letter(cs: Vec<char>, c: char) -> char {
        let i = cs.partition_point(|&x| x <= c);
        let idx = if i < cs.len() {i} else {0};
        unsafe {
            *cs.get_unchecked(idx)
        }
    }
}

